### PR TITLE
[SYCLomatic] Support migration of cusparseCsr2cscEx2[_bufferSize]

### PR DESCRIPTION
**Pull request by zhiweij1**
Signed-off-by: Jiang, Zhiwei <zhiwei.jiang@intel.com>

### CODE REVIEW

**Review by spencerpatty — APPROVED** [review]
omatcopy use for csr2csc looks good.  Approved

File: clang/test/dpct/cusparse-after101.cu [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  int* b_col_ptr;
+  int* b_row_ind;
+
+  size_t ws_size = 0;
+  void *ws;
+
+  // CHECK: ws_size = 0;
+  // CHECK-NEXT: dpct::sparse::csr2csc(*handle, 3, 4, a_val, a_row_ptr, a_col_ind, b_val, b_col_ptr, b_row_ind, dpct::library_data_t::real_float, oneapi::mkl::index_base::zero);
+  cusparseCsr2cscEx2_bufferSize(handle, 3, 4, 7, a_val, a_row_ptr, a_col_ind, b_val, b_col_ptr, b_row_ind, CUDA_R_32F, CUSPARSE_ACTION_NUMERIC, CUSPARSE_INDEX_BASE_ZERO, CUSPARSE_CSR2CSC_ALG1, &ws_size);
+  cusparseCsr2cscEx2(handle, 3, 4, 7, a_val, a_row_ptr, a_col_ind, b_val, b_col_ptr, b_row_ind, CUDA_R_32F, CUSPARSE_ACTION_NUMERIC, CUSPARSE_INDEX_BASE_ZERO, CUSPARSE_CSR2CSC_ALG1, ws);
```
Comment (zhimingwang36) [inline clang/test/dpct/cusparse-after101.cu:30]: what is the migration if CUSPARSE_ACTION_NUMERIC ==> CUSPARSE_ACTION_SYMBOLIC?

Comment (zhiweij1) [reply clang/test/dpct/cusparse-after101.cu:30]: done

**Comment by zhiweij1** [comment]
@spencerpatty , I further supported the migration of symbolic mode of cusparseCsr2cscEx2 by using a new array to save the output of the value array to keep the user's b_val is not touched.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**